### PR TITLE
[For Review] Update netcdf attrs for convention compliance

### PR DESCRIPTION
**Pull request by andrewdhicks**
Adds missing fields for CF and ACDD complicance.
Splits NCML util into separate file.

### CODE REVIEW

**Comment by andrewdhicks** [comment]
The cause of PQ failing compliance was missing `coverage_content_type` and `standard_name` for the variable.  These have been added, along with an attempt to fill in the `source` field.

I have not added the output of the ISO19115 mappings, which is as follows:

 ```
  date_created: '2017-07-07T03:43:26'
  date_modified: '2017-07-07T03:43:26'
  institution: Geoscience Australia - Client Services
  license: CC BY Attribution 4.0 International License
  product_version: Version 2.0, April 2015
  source: Please refer to the lineage section.
  summary: 'The PQ25 product facilitates interpretation and processing of Surface
    Reflectance (SR-N/NT), Fractional Cover 25 (FC25) and all derivative products.

    PQ25 is an assessment of each image pixel to determine if it is an unobscured,
    unsaturated observation of the Earth?s surface and also whether the pixel is represented
    in each spectral band. The PQ product allows users to produce masks which can
    be used to exclude pixels which don?t meet their quality criteria from analysis
    . The capacity to automatically exclude such pixels is essential for emerging
    multi-temporal analysis techniques that make use of every quality assured pixel
    within a time series of observations.


    Users can choose to process only land pixels, or only sea pixels depending on
    their analytical requirements, leading to enhanced computationally efficient.

    '
  time_coverage_start: '2017-07-07'
  title: Pixel Quality 25 v. 2.0.0
  uuid: 3db22d26-cd94-4b11-8108-74f84edf6346

```

The existing used fields are listed here:
https://github.com/GeoscienceAustralia/digitalearthau/blob/7041d9a4c1e00aa0d85e619bf2edf6f815ce3a34/ingestion/config/ls5_pq_albers.yaml#L12-L90

If there are any that need to be updated, let me know.

File: ingestion/config/ls5_nbar_albers.yaml [hunk 1]
```diff
@@ -73,6 +73,7 @@ global_attributes:
   publisher_email: earth.observation@ga.gov.au
   publisher_name: Section Leader, Operations Section, NEMO, Geoscience Australia
   publisher_url: 'http://www.ga.gov.au'
+  source: remotely observed surface reflectance
   license: CC BY Attribution 4.0 International License
   cdm_data_type: Grid
   product_suite: Surface Reflectance NBAR+ 25m
```
Comment (sixy6e) [inline ingestion/config/ls5_nbar_albers.yaml]: I think the 'product_suite' attribute for NBAR should exclude the '+' as it implies an additive extra

File: ingestion/config/ls7_nbar_albers.yaml [hunk 2]
```diff
@@ -73,6 +73,7 @@ global_attributes:
   publisher_email: earth.observation@ga.gov.au
   publisher_name: Section Leader, Operations Section, NEMO, Geoscience Australia
   publisher_url: 'http://www.ga.gov.au'
+  source: remotely observed surface reflectance
   license: CC BY Attribution 4.0 International License
   cdm_data_type: Grid
   product_suite: Surface Reflectance NBAR+ 25m
```
Comment (sixy6e) [inline ingestion/config/ls7_nbar_albers.yaml]: Same as before, removal of the '+' for the NBAR product.

File: ingestion/config/ls8_nbar_albers.yaml [hunk 3]
```diff
@@ -73,8 +73,8 @@ global_attributes:
   publisher_email: earth.observation@ga.gov.au
   publisher_name: Section Leader, Operations Section, NEMO, Geoscience Australia
   publisher_url: 'http://www.ga.gov.au'
+  source: remotely observed surface reflectance
   license: CC BY Attribution 4.0 International License
-  coverage_content_type: physicalMeasurement
   cdm_data_type: Grid
   product_suite: Surface Reflectance NBAR+ 25m
```
Comment (sixy6e) [inline ingestion/config/ls8_nbar_albers.yaml]: Removal of the '+' for the NBAR product.

File: ingestion/scripts/datacube-ingester [hunk 4]
```diff
@@ -96,7 +99,14 @@ def stack(config, year, queue, project, nodes, walltime, name):
     prep = 'datacube-stacker -v --app-config "%(config)s" --year %(year)s --save-tasks "%(taskfile)s"'
     cmd = prep % dict(config=config, taskfile=taskfile, year=year)
     if click.confirm('\n' + cmd + '\nRUN?', default=True):
-        subprocess.check_call(cmd, shell=True)
+        try:
+            subprocess.check_call(cmd, shell=True)
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 2:
+                click.echo('No tasks found - nothing to do!')
```
Comment (sixy6e) [inline ingestion/scripts/datacube-ingester:106]: I like :+1:

**Review by sixy6e — COMMENTED** [review]
Looks good.  A few items that pending a minor discussion may or may not need to be changes

**Pull request merged**